src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -266,14 +269,12 @@ def _normalize(
         # If user defined their udf as @udf(lambda: x, ...) then extension_fn.handler is <lambda>.
         extension_fn.handler = f"{py_file.stem}.{extension_fn.handler}"
 
-        snowpark_lib_found = False
-        snowpark_lib_regex = re.compile(SNOWPARK_LIB_REGEX)
-        for pkg in extension_fn.packages:
-            if snowpark_lib_regex.fullmatch(ensure_string_literal(pkg.strip())):
-                snowpark_lib_found = True
-                break
-        if not snowpark_lib_found:
-            extension_fn.packages.append(SNOWPARK_LIB_NAME)
+        extension_fn.packages = [
+            self._normalize_package_name(pkg) for pkg in extension_fn.packages
+        ]
+        snowpark_lib_name = ensure_string_literal(SNOWPARK_LIB_NAME)
```

Comment:
Seems like `ensure_string_literal(SNOWPARK_LIB_NAME)` is how the package is used everytime, as opposed to just `SNOWPARK_LIB_NAME`. Should line 45 then just be `SNOWPARK_LIB_NAME = ensure_string_literal("snowflake-snowpark-python")` so we don't have to keep track of the string representation?

Reply:
You missed the use in defining SNOWPARK_LIB_REGEX I think